src/sentry_core.c

```diff
@@ -197,9 +197,13 @@ sentry_shutdown(void)
             dumped_envelopes = sentry__transport_dump_queue(
                 options->transport, options->run);
         }
+        // crashpad can capture crashes *after* calling `shutdown`, so we leave
+        // the run folder intact so that it can still upload the msgpack files.
+#ifndef SENTRY_BACKEND_CRASHPAD
```

Comment:
I would prefer we check the registered and active backend instead of the define.  Maybe we can have a callback or flag on the backend instead.

Reply:
> the registered and active backend

its completely compile-time, there is no way to switch it at runtime, and sentry itself will not start if the backend fails to start.

Reply:
I know, but we already have a pattern for doing backend specific logic and that all checks against the backend on the options.